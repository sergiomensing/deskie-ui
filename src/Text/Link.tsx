import { AllHTMLAttributes, PropsWithChildren } from 'react';
import styled from 'styled-components';
import Text, { BaseTextProps } from './Text';

type LinkProps = AllHTMLAttributes<unknown> & BaseTextProps;

const StyledLink = styled(Text)`
  color: ${(props) => props.theme.colors.primary};

  &:hover {
    color: ${(props) => props.theme.colors.primaryHover};
  }
`;

function Link({ children, ...rest }: PropsWithChildren<LinkProps>) {
  return (
    <StyledLink as="a" {...rest}>
      {children}
    </StyledLink>
  );
}

export default Link;
