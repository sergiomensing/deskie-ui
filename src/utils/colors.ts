export const colorShade = (col: string, amt: number) => {
  col = col.replace(/^#/, '');
  if (col.length === 3) col = col[0] + col[0] + col[1] + col[1] + col[2] + col[2];

  const [sr, sg, sb] = col.match(/.{2}/g) || ['ff', 'ff', 'ff'];
  const [ar, ag, ab] = [parseInt(sr, 16) + amt, parseInt(sg, 16) + amt, parseInt(sb, 16) + amt];

  const r = Math.max(Math.min(255, ar), 0).toString(16);
  const g = Math.max(Math.min(255, ag), 0).toString(16);
  const b = Math.max(Math.min(255, ab), 0).toString(16);

  const rr = (r.length < 2 ? '0' : '') + r;
  const gg = (g.length < 2 ? '0' : '') + g;
  const bb = (b.length < 2 ? '0' : '') + b;

  return `#${rr}${gg}${bb}`;
};

const defaultColors = [
  '#E64980',
  '#4C6EF5',
  '#228BE6',
  '#15AABF',
  '#12B886',
  '#40C057',
  '#82C91E',
  '#FAB005',
  '#FD7E14',
  '#FA5252',
];

export function getColorFromString(value: string, colors: undefined | string[] = defaultColors) {
  let hash = 0;
  if (value.length === 0) return colors[hash];
  for (let i = 0; i < value.length; i++) {
    hash = value.charCodeAt(i) + ((hash << 5) - hash);
    hash = hash & hash;
  }
  hash = ((hash % colors.length) + colors.length) % colors.length;
  return colors[hash];
}
