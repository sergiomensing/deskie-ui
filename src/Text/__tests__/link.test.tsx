import React from 'react';
import Link from '../Link';
import { render } from '../../../tests/utils';

describe('Link', () => {
  it('should render <a> element', () => {
    const { container } = render(<Link>test-value</Link>);
    expect(container.querySelectorAll('a').length).not.toBe(0);
    expect(container).toMatchSnapshot();
  });
});
