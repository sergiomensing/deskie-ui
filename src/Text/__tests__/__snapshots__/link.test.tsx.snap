// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render <a> element 1`] = `
.c0 {
  color: #6C5DD3;
}

.c0:hover {
  color: #4535B7;
}

<div>
  <a
    class="c0"
  >
    test-value
  </a>
</div>
`;
